docs/docs/components-and-props.md

```diff
@@ -32,6 +32,8 @@ function Welcome(props) {
 
 This function is a valid React component because it accepts a single "props" object argument with data and returns a React element. We call such components "functional" because they are literally JavaScript functions.
 
+Note that the capitalization of `Welcome` does not indicate that it is a constructor. [It only indicates that the JSX tag is referring to a React component](https://facebook.github.io/react/docs/jsx-in-depth.html#html-tags-vs.-react-components).
```

Comment:
I mean that the whole convention of capitalizing constructors has no basis in the language. It's just a convention popularized by JSLint (or JSHint, I don't remember), I think. So for people not exposed to this convention (many React beginners) this implication will only further confuse them.